### PR TITLE
Vet-Clinic-Database: Feature/Database Performance Audit

### PR DESCRIPTION
**Pull request by zainsadaqat**
In this pull request, we have done the following:

- Improve performance for our database
- Insert Data
- Use `EXPLAIN ANALYZE`
- Perform `pair-programming`

1. Execution time for `SELECT COUNT(*) FROM visits where animal_id = 4;` before applying indexes

Screen Shot 2021-12-22 at 12 52 29 AM

2. Execution time for `SELECT COUNT(*) FROM visits where animal_id = 4;` after applying indexes

Screen Shot 2021-12-22 at 12 40 54 AM

3. Execution time for `SELECT * FROM visits where vet_id = 2;` before applying indexes

Screen Shot 2021-12-22 at 1 01 13 AM

4. Execution time for `SELECT * FROM visits where vet_id = 2;` after applying indexes

Screen Shot 2021-12-22 at 1 01 50 AM

5. Execution time for `SELECT * FROM owners where email = 'owner_18327@mail.com';` before applying indexes

Screen Shot 2021-12-22 at 1 02 43 AM

6. Execution time for `SELECT * FROM owners where email = 'owner_18327@mail.com';` after applying indexes

Screen Shot 2021-12-22 at 1 04 17 AM

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -112,4 +112,23 @@ CREATE TABLE VISITS(
...
+ALTER TABLE owners ADD COLUMN email VARCHAR(120);
+
+-- SELECT COUNT(*) FROM visits where animal_id = 4;
+CREATE INDEX ANIMALS_ID_INDEX ON VISITS(ANIMALS_ID);
+
+-- SELECT * FROM visits where vet_id = 2;
+CREATE INDEX VETS_ID_INDEX ON VISITS(VETS_ID);
+
+-- SELECT * FROM owners where email = 'owner_18327@mail.com';
+CREATE INDEX EMAIL_INDEX ON OWNERS(EMAIL);
```
Comment (descholar-ceo) [inline schema.sql:131]: - [x] You were supposed to include a pair of screenshots in the PR description with the results of the  EXPLAIN ANALYZE command for each of the above queries (in total it should be 6 screenshots):

:point_right: The first screenshot should show how much time the query takes BEFORE the improvements you introduced.
:point_right: The second screenshot should show how much time the query takes AFTER the improvements you introduced.

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
Hi @zainsadaqat and @mistermaxis , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (zainsadaqat) [reply schema.sql:131]: Really sorry for that, Completely forgot to attach the screenshots. Let me do it quickly.

**Comment by zainsadaqat** [comment]
## I have updated the pull request description by adding 6 screenshots before and after enhancing the performance. Kindly review our project.

**Review by descholar-ceo — APPROVED** [review]
Hi @zainsadaqat and @mistermaxis , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by zainsadaqat** [comment]
## Thank You so much @descholar-ceo for approving our pull request! 😃

**Pull request merged**